patches/api/0220-Add-methods-to-get-translation-keys.patch

```diff
@@ -393,6 +393,198 @@ index c799ac46dbe257d470d3b236cf55b41240f5fda8..d61855b83da0776fe910dee8cde184c7
      }
  
      // Paper start - Add villager reputation API
+diff --git a/src/main/java/org/bukkit/event/inventory/InventoryType.java b/src/main/java/org/bukkit/event/inventory/InventoryType.java
+index e1e7f45fd0f65d0874dd0698da436c7ac2e7951b..51e0ac7f0751a51182233f15dbd9229cf68b1685 100644
+--- a/src/main/java/org/bukkit/event/inventory/InventoryType.java
++++ b/src/main/java/org/bukkit/event/inventory/InventoryType.java
+@@ -19,121 +19,121 @@ import org.jetbrains.annotations.NotNull;
+  *
+  * @see org.bukkit.Bukkit#createInventory(InventoryHolder, InventoryType)
+  */
+-public enum InventoryType {
++public enum InventoryType implements net.kyori.adventure.translation.Translatable { // Paper - translations
+ 
+     /**
+      * A chest inventory, with 0, 9, 18, 27, 36, 45, or 54 slots of type
+      * CONTAINER.
+      */
+-    CHEST(27, "Chest"),
++    CHEST(27, "Chest", "container.chest"), // Paper
+     /**
+      * A dispenser inventory, with 9 slots of type CONTAINER.
+      */
+-    DISPENSER(9, "Dispenser"),
++    DISPENSER(9, "Dispenser", "container.dispenser"), // Paper
+     /**
+      * A dropper inventory, with 9 slots of type CONTAINER.
+      */
+-    DROPPER(9, "Dropper"),
++    DROPPER(9, "Dropper", "container.dropper"), // Paper
+     /**
+      * A furnace inventory, with a RESULT slot, a CRAFTING slot, and a FUEL
+      * slot.
+      */
+-    FURNACE(3, "Furnace"),
++    FURNACE(3, "Furnace", "container.furnace"), // Paper
+     /**
+      * A workbench inventory, with 9 CRAFTING slots and a RESULT slot.
+      */
+-    WORKBENCH(10, "Crafting"),
++    WORKBENCH(10, "Crafting", "container.crafting"), // Paper
+     /**
+      * A player's crafting inventory, with 4 CRAFTING slots and a RESULT slot.
+      * Also implies that the 4 ARMOR slots are accessible.
+      */
+-    CRAFTING(5, "Crafting", false),
++    CRAFTING(5, "Crafting", false, "container.crafting"), // Paper
+     /**
+      * An enchantment table inventory, with two CRAFTING slots and three
+      * enchanting buttons.
+      */
+-    ENCHANTING(2, "Enchanting"),
++    ENCHANTING(2, "Enchanting", "container.enchant"), // Paper
+     /**
+      * A brewing stand inventory, with one FUEL slot and four CRAFTING slots.
+      */
+-    BREWING(5, "Brewing"),
++    BREWING(5, "Brewing", "container.brewing"), // Paper
+     /**
+      * A player's inventory, with 9 QUICKBAR slots, 27 CONTAINER slots, 4 ARMOR
+      * slots and 1 offhand slot. The ARMOR and offhand slots may not be visible
+      * to the player, though.
+      */
+-    PLAYER(41, "Player"),
++    PLAYER(41, "Player", "container.crafting"), // Paper
```

Comment:
I think this should be null, the survival crafting slots is not even mentioned in the javadoc, and the whole inventory doesn't have a name. Additionally those 4 crafting slots gets the right CRAFTING type and not PLAYER.

Reply:
I use that, cause that's what is used on the client for the player's inventory screen.